src/compute/src/logging/initialize.rs

```diff
@@ -88,31 +91,40 @@ struct LoggingContext<'a, A: Allocate> {
 }
 
 impl<A: Allocate + 'static> LoggingContext<'_, A> {
-    fn construct_dataflows(&mut self) -> BTreeMap<LogVariant, TraceBundle> {
-        let mut traces = BTreeMap::new();
-        traces.extend(super::timely::construct(
+    fn construct_dataflows(&mut self) -> BTreeMap<LogVariant, (TraceBundle, usize)> {
+        let mut dataflows = BTreeMap::new();
+
+        let index = self.worker.next_dataflow_index();
```

Comment:
This assumes that each `construct` call only creates a single dataflow. Would it be nicer if `construct` returned `(LogVariant, (usize, RowRowAgent<...>, Rc<...>))` to encode the dataflow id per log variant?

Reply:
Maybe! Probably a good opportunity to introduce a struct to replace the tuple too.